File: backend/src/modules/chat/entities/chat_user.entity.ts
import { ChatRole } from "../enums/chat_role.enum";
import { User } from "@prisma/client";

export class ChatUserEntity {
  id: string;
  name: string;
  email: string;
  role: ChatRole;
  pictureUrl?: string;
  createdAt: Date;

  constructor(
    id: string,
    name: string,
    email: string,
    role: ChatRole,
    pictureUrl?: string,
    createdAt: Date = new Date()
  ) {
    this.id = id;
    this.name = name;
    this.email = email;
    this.pictureUrl = pictureUrl;
    this.createdAt = createdAt;
    this.role = role;
  }

  static fromPrisma(user: User, role: ChatRole): ChatUserEntity {
    return new ChatUserEntity(
      user.id,
      user.name,
      user.email,
      role,
      user.picture_url,
      user.createdAt
    );
  }
}

File: frontend/falatu_mobile/lib/chat/core/domain/entities/chat/chat_user_entity.dart
import "package:equatable/equatable.dart";
import "package:falatu_mobile/chat/utils/enums/chat_role.dart";

class ChatUserEntity extends Equatable {
  final String id;
  final String name;
  final String email;
  final String? pictureUrl;
  final DateTime createdAt;
  final ChatRole role;

  const ChatUserEntity(
      {required this.id,
      required this.name,
      required this.email,
      required this.pictureUrl,
      required this.createdAt,
      required this.role});

  @override
  List<Object?> get props => [id, name, pictureUrl, createdAt, role];
}
